### PR TITLE
unit tests for design examples

### PR DESCRIPTION
**Pull request by alex-golts**

### CODE REVIEW

File: fusedrug_examples/requirements.txt [hunk 1]
```diff
@@ -5,8 +5,7 @@ dgl
...
 deepchem
 mol2vec @ git+https://github.com/samoturk/mol2vec
-
-# new depdencies for design/amp example
 modlamp
 torchtext==0.3.1 # old version of tokenizer - need to replace with another tokenizer and remove this
-biopython
\ No newline at end of file
+biopython
+x_transformers
```
Comment (mosheraboh) [inline fusedrug_examples/requirements.txt:11]: Did I use x_transformers? where?

File: fusedrug_examples/tests/test_design_amp.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import fusedrug_examples.design.amp.classifier.main_classifier_train as main_classifier_train
+import fusedrug_examples.design.amp.design.main_design_train as main_design_train
+from omegaconf import OmegaConf
+from pathlib import Path
+class DesignAMPTestCase(unittest.TestCase):
+
+    def test_classifier(self):
+        config_path = Path(__file__, "../../design/amp/classifier/config.yaml")
+        cfg = OmegaConf.load(config_path)
+        cfg.train.trainer_kwargs.max_epochs = 1
```
Comment (mosheraboh) [inline fusedrug_examples/tests/test_design_amp.py:11]: Not sure how much the test takes
But if you case load just part of the data and not all of it.
You can do that be setting uniprot data path to None.

**Review by mosheraboh — APPROVED** [review]
Great!
Hope creating the test was smooth

**Comment by mosheraboh** [comment]
Anything was missing or not clear in the instructions? 
Maybe we should also mention the env var of the split file?

File: fusedrug_examples/requirements.txt (hunk 1 above)
Comment (alex-golts) [reply fusedrug_examples/requirements.txt:11]: you import things from `fuse.dl.models.backbones.backbone_transformer` which imports it. here for example:
https://github.com/BiomedSciAI/fuse-drug/blob/c9ea278c1883bea65105cbbb26d39b07ea07e965/fusedrug_examples/design/amp/classifier/main_classifier_train.py#L42

File: fusedrug_examples/tests/test_design_amp.py (hunk 2 above)
Comment (alex-golts) [reply fusedrug_examples/tests/test_design_amp.py:11]: done.

**Comment by alex-golts** [comment]
> Anything was missing or not clear in the instructions?
> Maybe we should also mention the env var of the split file?

I added instructions for it now. I found it a bit confusing that an output file is specified via environment variable. maybe it would be better to have it under the config.

besides that the instructions are pretty clear and code looks really well organized too! I have to say I somehow missed the readme until now and was still able to figure out most of how to set the paths from the code, so I guess that's good :-)

File: fusedrug_examples/requirements.txt (hunk 1 above)
Comment (mosheraboh) [reply fusedrug_examples/requirements.txt:11]: If I'm not missing anything,  it means that fuse is dependent on x_transformer.
This dependency is already isted here: https://github.com/BiomedSciAI/fuse-med-ml/blob/f955b484a7bec89a7a6ca27e12dfcb400353c984/fuse/requirements.txt#L32 
(to resolve you probably just needed to rerun fuse installation to get the new dependencies).

**Comment by mosheraboh** [comment]
> > Anything was missing or not clear in the instructions?
> > Maybe we should also mention the env var of the split file?
> 
> I added instructions for it now. I found it a bit confusing that an output file is specified via environment variable. maybe it would be better to have it under the config.
> 
> besides that the instructions are pretty clear and code looks really well organized too! I have to say I somehow missed the readme until now and was still able to figure out most of how to set the paths from the code, so I guess that's good :-)

Glad to hear! Thanks for the modifications.

**Pull request merged**